java/_posts/2016-07-15-create-java-dockerfile.markdown

```diff
@@ -0,0 +1,486 @@
+
+```bash
+PASS=${TOMCAT_PASS:-$(pwgen -s 12 1)}
+_word=$( [ ${TOMCAT_PASS} ] && echo "preset" || echo "random" )
+
+echo "=> Creating an admin user with a ${_word} password in Tomcat"
+sed -i -r 's/<\/tomcat-users>//' ${CATALINA_HOME}/conf/tomcat-users.xml
+echo '<role rolename="manager-gui"/>' >> ${CATALINA_HOME}/conf/tomcat-users.xml
+echo '<role rolename="manager-script"/>' >> ${CATALINA_HOME}/conf/tomcat-users.xml
+echo '<role rolename="manager-jmx"/>' >> ${CATALINA_HOME}/conf/tomcat-users.xml
+echo '<role rolename="admin-gui"/>' >> ${CATALINA_HOME}/conf/tomcat-users.xml
+echo '<role rolename="admin-script"/>' >> ${CATALINA_HOME}/conf/tomcat-users.xml
+echo "<user username=\"admin\" password=\"${PASS}\" roles=\"manager-gui,manager-script,manager-jmx,admin-gui, admin-script\"/>" >> ${CATALINA_HOME}/conf/tomcat-users.xml
+echo '</tomcat-users>' >> ${CATALINA_HOME}/conf/tomcat-users.xml
+echo "=> Done!"
+touch /.tomcat_admin_created
+
+echo "========================================================================"
+echo "You can now configure to this Tomcat server using:"
+echo ""
+echo "    admin:${PASS}"
+echo ""
+echo "========================================================================"
+```   
+
+This file creates the Tomcat admin user.
+Add one more file in the same directory named as `run.sh` with following content. This will call the create users file and then reload the Tomcat server.
+
+```bash
+!/bin/bash
+
+if [ ! -f /.tomcat_admin_created ]; then
+    /create_tomcat_admin_user.sh
+fi
+
+exec ${CATALINA_HOME}/bin/catalina.sh run
+```
+
+### 5. Build and Test the Docker Image
+
+Build the docker image using following command.
+
+```bash
+$ docker build -f Dockerfile -t demo/spring:maven-3.3-jdk-8 .
+```
+
+Test the image.
+
+[Here](https://github.com/atifsaddique211f/spring-maven-sample.git) is the sample Spring Maven-based project that you can use for testing. You can download the zip or clone it by using the following command:
+
+```bash
+$ git clone  https://github.com/atifsaddique211f/spring-maven-sample.git
+```
+
+Go to the project directory:
+
+```bash
+$ cd spring-maven-sample
+```
+
+Run the following command to build and package the project:
+
+```bash
+$ docker run -it --rm -v "$PWD":/app -w /app demo/spring:maven-3.3-jdk-8 mvn clean install
+```
+
+The above command will create a war file at `target/springwebapp.war`.
+
+Copy this war file into the Tomcat *webapps* directory:
+
+```bash
+$ docker run -it -d --name spring -p 8080:8080 -v "$PWD":/app demo/spring:maven-3.3-jdk-8 bash -c "cp /app/target/springwebapp.war /tomcat/webapps/ & /tomcat/bin/catalina.sh run"
+```
+
+The above command runs a container named `spring` in detached mode. It also provides mapping from the Docker container port `8080` to host machine port `8080`.
+
+You can access your web application at the following url in the browser.
+[http://localhost:8080/springwebapp/car/add](http://localhost:8080/springwebapp/car/add).
```

Comment:
That was intentional because I figured the links would work if the user were following along in the tutorial.